ERCS/erc-7524.md

```diff
@@ -0,0 +1,211 @@
+
+$c == \text{hash}(\text{nul}, g^r, h^r)$
+
+Due to SHA-256 being a native precompile on Ethereum, this operation will still be efficient for smart contract verifiers.
+
+### Version 3: 
+
+There may be a more efficient V3 in the future, perhaps via removing indifferentiability from hash_to_curve.
+
+## Rationale
+
+We will define a few specific properties we are looking for in a candidate algorithm, then define a few other intuitive algorithms and explain why they don’t actually work.
+
+- Noninteractivity
+    - The importance of noninteractivity in ZK ID systems is that it enables a large anonymity set from the start, making it resistant to sybil attacks and spam, which would be possible if there was an interactive phase. This allows for new use cases such as ZK airdrops.
+    - Noninteractivity enables the full set of eligible users to be part of the anonymity set, without requiring any interaction. This is possible if the zk proof can verify the set membership in the Merkle tree, the message via the signature, and the unique nullifier. Interactive nullifiers, such as tornado.cash's, require updating the anonymity set Merkle tree with each new user,
+- Uniqueness
+    - If we want to forbid actions like double spending or double claiming, we need them to be verifiably unique per account.
+    - For example: Because ECDSA signatures are nondeterministic, signatures don’t suffice; we need a new deterministic function, verifiable with only the public key. We want the nullifier to be non-interactive, to uniquely identify the keypair yet keep the account identity secret.
+    - The key insight is that such nullifiers can be used as a public commitment to a specific anonymous account to provide us with a uniqueness guarantee.
+- Deterministic
+    - We want each account to only generate one such signature, and generate it exactly the same over time into the future.
+- Verifiable without a secret key
+    - In cases where signatures are nondeterministic (like ECDSA) the signature alone is not sufficient for verification.
+    - We want a new, deterministic function verifiable only with the public key
+    - We don’t want users copy-pasting secret keys anywhere, and we need to choose a function such that the enclave calculation is simple enough for hardware wallets.
+    - Because the nullifier is non-interactive, we are able to uniquely identify the key pair without revealing the account identity.
+
+For a few possible simpler algorithm designs that were considered, see blog.aayushg.com > /posts/plume > ‘**One address <-> one nullifier’** section. We based the final design off of BLS signatures, Chaum-Pederson EQDL, and Goh-Jarecki’s EDL paper, but to work on secp256k1.
+
+## Reference Implementation
+
+The GitHub repository `plume-sig/zk-nuliifier-sig` contains the reference implementation in several languages including Rust and Javascript. There is a sample PR to Metamask Core as well, and a PR to Taho Core incoming.
```

Comment:
```suggestion
```

Either include a reference implementation here (or in your assets folder), or remove this section entirely.